bson/binary.py

```diff
@@ -247,6 +247,9 @@ def __init__(self, data: Sequence[float | int], dtype: BinaryVectorDtype, paddin
         self.dtype = dtype
         self.padding = padding
 
+    def __repr__(self) -> str:
+        return f"BinaryVector - {self.dtype=}, {self.dtype},{self.padding=}, {self.data=}"
```

Comment:
This needs to follow Python's convention for repr: https://docs.python.org/3/library/functions.html#repr

And we should add tests for this like we do for our other custom type reprs.